lib/algora/admin.ex

```diff
@@ -16,6 +16,119 @@ defmodule Algora.Admin do
+        case x do
+          %ExM3U8.Tags.MediaInit{uri: uri} -> [
+            %ExM3U8.Tags.MediaInit{uri: Storage.to_absolute(:video, video.uuid, uri)}
+            | acc
+          ]
+          %ExM3U8.Tags.Segment{uri: uri, duration: duration} -> [
+            %ExM3U8.Tags.Segment{uri: Storage.to_absolute(:video, video.uuid, uri), duration: duration}
+            | acc
+          ]
+          others -> [others | acc]
+        end
+      end)
+    |> Enum.reverse()
+
+    %ExM3U8.MediaPlaylist{timeline: timeline, info: info}
+  end
+
+  defp merge_media_playlists(video, nil), do: get_absolute_media_playlist(video)
+
+  defp merge_media_playlists(video, playlist) do
+    new_playlist = video |> get_absolute_media_playlist
+
+    %ExM3U8.MediaPlaylist{
+      playlist
+      | timeline: playlist.timeline ++ [%ExM3U8.Tags.Discontinuity{} | new_playlist.timeline],
+      info: %ExM3U8.MediaPlaylist.Info{
+        playlist.info
+        | target_duration: max(playlist.info.target_duration, new_playlist.info.target_duration)
+      }
+    }
+  end
+
+  defp merge_playlists(videos) do
+    playlists = Enum.map(videos, &get_playlist/1)
+
+    streams = Enum.flat_map(playlists, &Map.get(&1, :items))
+    example_stream = Enum.at(streams, 1)
+
+    if Enum.all?(streams, fn x -> example_stream.resolution == x.resolution && example_stream.codecs == x.codecs end) do
+      max_bandwidth = Enum.map(streams, &Map.get(&1, :bandwidth)) |> Enum.max(&Ratio.gte?/2)
+      avg_bandwidth = streams
+        |> Enum.reduce(0, fn s, sum -> sum + s.average_bandwidth end)
+        |> then(&(&1 / Enum.count(streams)))
+
+      {:ok, %{ Enum.at(playlists, 0)
+          | items: [
+            %{ example_stream
+              | average_bandwidth: Ratio.trunc(avg_bandwidth),
+              bandwidth: max_bandwidth
+            }
+          ]
+        }
+      }
+    else
+      {:error, "Codecs or resolutions don't match"}
+    end
+  end
+
+  defp copy_index_to_video(from_uuid, to_uuid) do
+    ExAws.S3.put_object_copy(
+      Storage.bucket(),
+      "#{to_uuid}/index.m3u8",
+      Storage.bucket(),
+      "#{from_uuid}/index.m3u8"
+    ) |> ExAws.request()
+  end
+
+  defp insert_merged_video(videos, playlist, media_playlist) do
+    [video | _] = videos
+
+    duration = videos |> Enum.reduce(0, fn v, d -> d + v.duration end)
+
+    result = %{video | duration: duration, id: nil, filename: nil}
+      |> change()
+      |> Video.put_video_url(video.format)
+      |> Repo.insert()
+
+    upload_to = fn uuid, track_atom, m3u8_body -> Storage.upload(
+      "#{ExM3U8.serialize(m3u8_body)}#EXT-X-ENDLIST\n",
+      "#{uuid}/#{@tracks[track_atom]}",
+      content_type: "application/x-mpegURL"
+    ) end
+
+    with {:ok, new_video} <- result,
+      {:ok, _} <- upload_to.(new_video.uuid, :manifest, playlist),
+      {:ok, _} <- upload_to.(new_video.uuid, :video, media_playlist),
+      {:ok, _} <- copy_index_to_video(video.uuid, new_video.uuid)do
```

Comment:
Do we need this still? I think it's overwriting the previously generated playlist

Reply:
Whoops, you're right. Fixed!